cli/doc/printer/mod.rs

```diff
@@ -0,0 +1,11 @@
+use super::DocNode;
+
+mod json;
+mod terminal;
+pub use json::JSONPrinter;
+pub use terminal::TerminalPrinter;
+
+pub trait Printer {
+  fn print(&self, doc_nodes: Vec<DocNode>);
+  fn print_details(&self, node: DocNode);
+}
```

Comment:
Is this trait useful? Can we not just combine this whole `cli/doc/printer` subtree into a single `cli/doc/printer.rs`? It seems like an unnecessary abstraction.

Neither of these printers implement the `Display` trait in Rust which allows them to be written to different outputs. Here, presumably, they all write to stdout?